src/lib/InstantSearch.ts

```diff
@@ -323,30 +354,31 @@ See: https://www.algolia.com/doc/guides/building-search-ui/going-further/backend
     // The helper needs to be reset to perform the next search from a fresh state.
     // If not reset, it would use the state stored before calling `dispose()`.
     this.removeAllListeners();
-    this.mainHelper.removeAllListeners();
+    this.mainHelper!.removeAllListeners();
     this.mainHelper = null;
     this.helper = null;
   }
 
-  scheduleSearch = defer(() => {
-    this.mainHelper.search();
+  public scheduleSearch = defer(() => {
+    this.mainHelper!.search();
   });
 
-  scheduleRender = defer(() => {
-    if (!this.mainHelper.hasPendingRequests()) {
+  public scheduleRender = defer(() => {
+    if (this.mainHelper!.hasPendingRequests() === false) {
       clearTimeout(this._searchStalledTimer);
       this._searchStalledTimer = null;
       this._isSearchStalled = false;
     }
 
-    this.mainIndex.render({
+    // @TODO: do we want this to be correct arguments?
+    this.mainIndex.render(({
       instantSearchInstance: this,
-    });
+    } as any) as RenderOptions);
```

Comment:
here also, we call this with less than what index asks for

Reply:
The suggestion for `init` also apply for `render`.